client/environment.go

```diff
@@ -1,7 +1,38 @@
 package client
 
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
 type ConfigurationVariableType int
 
+func (c *ConfigurationVariableType) ReadResourceData(fieldName string, d *schema.ResourceData) error {
```

Comment:
a new way to serialize custom fields between terraform and golang structs.
A little bit complex, but hopefully clear enough.